api/handlers.py

```diff
@@ -1,22 +1,44 @@
 import webapp2
-from google.appengine.ext.webapp import template
 from loveutils import messages
 from loveutils import config
+from webapp2_extras import jinja2
 
 
-MESSAGE_TEMPLATE_PATH = 'templates/message.html'
-HOME_TEMPLATE_PATH = 'templates/home.html'
-CREATE_TEMPLATE_PATH = 'templates/create.html'
-CREATE_GUIDED_TEMPLATE_PATH = 'templates/create-guided.html'
+MESSAGE_TEMPLATE_PATH = 'message.html'
+HOME_TEMPLATE_PATH = 'home.html'
+CREATE_TEMPLATE_PATH = 'create.html'
+CREATE_GUIDED_TEMPLATE_PATH = 'create-guided.html'
 MESSAGES = messages.MESSAGES
 
 
-def handle_404():
-    template_values = {
-        'message': '404 not found. Oh NO! Nobody loves a 404!',
-        'signature': 'The LoveMaster'
-    }
-    return template.render(MESSAGE_TEMPLATE_PATH, template_values)
+class BaseHandler(webapp2.RequestHandler):
+    @webapp2.cached_property
+    def jinja2(self):
+        # Returns a Jinja2 renderer cached in the app registry.
+        return jinja2.get_jinja2(app=self.app)
+
+    def render_response(self, _template, **context):
+        # Renders a template and writes the result to the response.
+        rv = self.jinja2.render_template(_template, **context)
+        self.response.write(rv)
+
+
+class Handler404(BaseHandler):
+    def get(self):
+        template_values = {
+            'message': '404 not found. Oh NO! Nobody loves a 404!',
+            'signature': 'The LoveMaster'
+        }
+        return self.render_response(MESSAGE_TEMPLATE_PATH, **template_values)
```

Comment:
Even if it work – not sure how though, `self.render_response(…)` returns `None`, so returning here is pointless.
I would just remove this `return` statement or return a `webapp2.Response(…)` in `def render_response`.